sklearn/preprocessing/_polynomial.py

```diff
@@ -346,7 +364,18 @@ def transform(self, X):
         for i in range(n_features):
             spl = self.bsplines_[i]
 
-            if self.extrapolation in ("continue", "error"):
+            if self.extrapolation in ("continue", "error", "periodic"):
+
+                if self.extrapolation == "periodic":
+                    # With periodic extrapolation we map x to the segment
+                    # [spl.t[k], spl.t[n]].
+                    # This is equivalent to BSpline(.., extrapolate="periodic")
+                    # for scipy>=1.6.1.
+                    n = spl.t.size - spl.k - 1
+                    X[:, i] = spl.t[spl.k] + (X[:, i] - spl.t[spl.k]) % (
+                        spl.t[n] - spl.t[spl.k]
+                    )
```

Comment:
Maybe you can add a test that checks that you can reproduce the same transformation with scipy >= 1.6.1 and `pytest.skip` otherwise.

Reply:
I'm not sure how to implement this. We could do some monkeypatching. Can you point me to such a test for a similar setting?

Reply:
Write a test that compares the out-of-range prediction of a `SplineTransformer(extrapolation="periodic")` transformer with the matching transformation computed using `BSpline(.., extrapolate="periodic")` from scipy on the same data (using a dataset with a single feature).

Reply:
Use only 1 or 2 knots to make the test as simple as possible.

Reply:
The minimal example is degree 2 with three knots. I've added a test.